Training-27: Hadoop Training Slides

greatgautam opened the pull request:
Create Apache Hadoop Slides for adding to Apache Training

RyanSkraba commented:
Hello!  I think you accidently merged the proposed TRAINING-28 (Hive) slides into the TRAINING-27 (Hadoop) PR.

I don't know if we have best practices yet -- but I find it very useful to have the JIRA link in the PR name (and optionally in the commit messages).

Can you remove the `.DS_Store` files from the PR?

greatgautam commented on content/ApacheHadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,80 @@
+- one of more of DataNodes used for storage and serving read and write requests from the file system’s clients
+
+== HDFS New Features
+New features and improvements are regularly implemented in HDFS. The following is a subset of useful features in HDFS:
```
I have updated it.

greatgautam commented on content/ApacheHadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,80 @@
+(input) <k1, v1> -> map -> <k2, v2> -> combine -> <k2, v2> -> reduce -> <k3, v3> (output)
+
+== HDFS Architecture
+HDFS has a master/slave architecture. An HDFS cluster consists of:
```
Added new line.

greatgautam commented on content/ApacheHadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,80 @@
+
+Hadoop MapReduce is a software framework for easily writing applications which process vast amounts of data (multi-terabyte data-sets) in-parallel on large clusters (thousands of nodes) of commodity hardware in a reliable, fault-tolerant manner. 
+
+The MapReduce framework consists of:
```
Added a newline. Please confirm if it looks good.

greatgautam commented:
Hi Ryan,
Thanks. I have added commands as well. Please review.

regards,
Gautam

On Fri, Dec 6, 2019 at 3:03 AM RyanSkraba <notifications@github.com> wrote:

> *@RyanSkraba* commented on this pull request.
>
> Hey, sorry for not coming back to this and thanks for addressing all the
> non-content points I raised.
>
> I actually thought the content was still ongoing! (I added a comment
> pointing to an empty slide)
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r354771610>
> :
>
> > +== HDFS Checkpoint node
> +
> +Checkpoint node performs periodic checkpoints of the namespace.
> +It helps minimize the size of the log stored at the NameNode containing changes to the HDFS.
> +It replaces the role previously filled by the Secondary NameNode.
> +The NameNode allows multiple Checkpoint nodes simultaneously, as long as there are no Backup nodes registered with the system.
> +
> +
> +== HDFS Backup node
> +
> +Backup node is an extension to the Checkpoint node.
> +In addition to checkpointing it also receives a stream of edits from the NameNode.
> +It maintains its own in-memory copy of the namespace, which is always in sync with the active NameNode namespace state.
> +Only one Backup node may be registered with the NameNode at once.
> +
> +== HDFS Commands
>
> Empty slide!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-training/pull/63?email_source=notifications&email_token=AA5KNGYFVHF7NZAADMNEVPDQXIWPLA5CNFSM4JKQDJYKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCOHM7FI#pullrequestreview-328126357>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5KNG2N5BH3ZY6536SJF33QXIWPLANCNFSM4JKQDJYA>
> .
>

RyanSkraba commented:
It builds correctly, and the slides look correctly formatted!  I agree this could serve as a starting point for someone to build on -- as I mentioned, Hadoop is a vast subject!

Calling any committer to take a look :D

Have you been following this https://github.com/apache/incubator-training/pull/43 ?

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+
+== What is Apache Hadoop?
+
+Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
```
Updated.

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
+
+Two main layers:
+
```
I have added the new text.

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+
+The MapReduce framework consists of:
+
+- single master ResourceManager
```
Updated!

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+
+Hadoop Distributed File System (HDFS) is a distributed file system designed to run on commodity hardware. 
+
+- similar to other distributed file systems
```
Thanks, I took care of it/

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+- Processing layer (MapReduce)
+- Storage layer (Hadoop Distributed File System)
+
+== MapReduce
```
Updated

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+
+== HDFS Special Nodes
+
+Main nodes in HDFS are as follows:
```
Thanks, I took care of it.

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+== HDFS Architecture
+
+HDFS has a master/slave architecture. An HDFS cluster consists of:
+
```
Added the diagram.

greatgautam commented on content/Hadoop/src/main/asciidoc/index.adoc:
```diff
@@ -0,0 +1,126 @@
+Usage: hdfs [SHELL_OPTIONS] COMMAND [GENERIC_OPTIONS] [COMMAND_OPTIONS]
+
+
+== Common HDFS Commands
```
Yes, added for YARN.

greatgautam commented:
Hi,
I have updated the PR for Apache Hadoop slides. Please review and merge if
it looks good.

https://github.com/apache/incubator-training/pull/63

thanks,
Gautam

On Fri, Jan 10, 2020 at 1:54 AM Brahma Reddy Battula <
notifications@github.com> wrote:

> *@brahmareddybattula* commented on this pull request.
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363564712>
> :
>
> > +      http://www.apache.org/licenses/LICENSE-2.0
>
> +
>
> +  Unless required by applicable law or agreed to in writing, software
>
> +  distributed under the License is distributed on an "AS IS" BASIS,
>
> +  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
>
> +  See the License for the specific language governing permissions and
>
> +  limitations under the License.
>
> +
>
> +////
>
> +:revealjs_progress: true
>
> +:revealjs_slidenumber: true
>
> +:sourcedir: ../java
>
> +
>
> +== What is Apache Hadoop?
>
> +
>
> +Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
>
>
> I feel, We can describe like below.
>
> The Apache Hadoop software library is a framework that allows for the
> distributed processing of large data sets across clusters of computers
> using simple programming models. It is designed to scale up from single
> servers to thousands of machines, each offering local computation and
> storage. Rather than rely on hardware to deliver high-availability, the
> library itself is designed to detect and handle failures at the application
> layer, so delivering a highly-available service on top of a cluster of
> computers, each of which may be prone to failures.
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363566531>
> :
>
> > +  distributed under the License is distributed on an "AS IS" BASIS,
>
> +  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
>
> +  See the License for the specific language governing permissions and
>
> +  limitations under the License.
>
> +
>
> +////
>
> +:revealjs_progress: true
>
> +:revealjs_slidenumber: true
>
> +:sourcedir: ../java
>
> +
>
> +== What is Apache Hadoop?
>
> +
>
> +Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
>
> +
>
> +Two main layers:
>
> +
>
>
> The base Apache Hadoop framework is composed of the following modules:
>
>
>
> Hadoop Common – contains libraries and utilities needed by other Hadoop modules;
>
> Hadoop Distributed File System (HDFS) – a distributed file-system that stores data on commodity machines, providing very high aggregate bandwidth across the cluster;
>
> Hadoop YARN – (introduced in 2012) a platform responsible for managing computing resources in clusters and using them for scheduling users' applications;[10][11]
>
> Hadoop MapReduce – an implementation of the MapReduce programming model for large-scale data processing.
>
>
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363566774>
> :
>
> > +== What is Apache Hadoop?
>
> +
>
> +Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
>
> +
>
> +Two main layers:
>
> +
>
> +- Processing layer (MapReduce)
>
> +- Storage layer (Hadoop Distributed File System)
>
> +
>
> +== MapReduce
>
> +
>
> +Hadoop MapReduce is a software framework for easily writing applications which process vast amounts of data (multi-terabyte data-sets) in-parallel on large clusters (thousands of nodes) of commodity hardware in a reliable, fault-tolerant manner.
>
> +
>
> +The MapReduce framework consists of:
>
> +
>
> +- single master ResourceManager
>
>
> This is for YARN, not for MR.
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363567463>
> :
>
> > +
>
> +== MapReduce
>
> +
>
> +Hadoop MapReduce is a software framework for easily writing applications which process vast amounts of data (multi-terabyte data-sets) in-parallel on large clusters (thousands of nodes) of commodity hardware in a reliable, fault-tolerant manner.
>
> +
>
> +The MapReduce framework consists of:
>
> +
>
> +- single master ResourceManager
>
> +- one worker NodeManager per cluster-node
>
> +- MRAppMaster per application
>
> +
>
> +== Hadoop Distributed File System
>
> +
>
> +Hadoop Distributed File System (HDFS) is a distributed file system designed to run on commodity hardware.
>
> +
>
> +- similar to other distributed file systems
>
>
> I suggest, you go through the following and update the same.
>
>
> https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363567669>
> :
>
> > +
>
> +////
>
> +:revealjs_progress: true
>
> +:revealjs_slidenumber: true
>
> +:sourcedir: ../java
>
> +
>
> +== What is Apache Hadoop?
>
> +
>
> +Apache Hadoop is a framework that allows for the distributed processing of large data sets across clusters of computers using simple programming models.
>
> +
>
> +Two main layers:
>
> +
>
> +- Processing layer (MapReduce)
>
> +- Storage layer (Hadoop Distributed File System)
>
> +
>
> +== MapReduce
>
>
> I suggest, you go through the following and update.
>
>
> https://hadoop.apache.org/docs/stable/hadoop-mapreduce-client/hadoop-mapreduce-client-core/MapReduceTutorial.html
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363567952>
> :
>
> > +== HDFS Useful Features
>
> +
>
> +New features and improvements are regularly implemented in HDFS. The following is a subset of useful features in HDFS:
>
> +
>
> +- File permissions and authentication.
>
> +- Rack awareness: to take a node’s physical location into account while scheduling tasks and allocating storage.
>
> +- Safemode: an administrative mode for maintenance.
>
> +- fsck: a utility to diagnose health of the file system, to find missing files or blocks.
>
> +- fetchdt: a utility to fetch DelegationToken and store it in a file on the local system.
>
> +- Balancer: tool to balance the cluster when the data is unevenly distributed among DataNodes.
>
> +- Upgrade and rollback: after a software upgrade, it is possible to rollback to HDFS’ state before the upgrade in case of unexpected problems.
>
> +
>
> +
>
> +== HDFS Special Nodes
>
> +
>
> +Main nodes in HDFS are as follows:
>
>
> Include high availablity, these nodes are almost not used now. We can try
> to updated data...?
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363568291>
> :
>
> > +== MapReduce Inputs and Outputs
>
> +
>
> +The MapReduce framework operates exclusively on <key, value> pairs, that is, the framework views the input to the job as a set of <key, value> pairs and produces a set of <key, value> pairs as the output of the job, conceivably of different types.
>
> +
>
> +The key and value classes have to be serializable by the framework and hence need to implement the Writable interface. Additionally, the key classes have to implement the WritableComparable interface to facilitate sorting by the framework.
>
> +
>
> +Input and Output types of a MapReduce job:
>
> +
>
> +(input) <k1, v1> -> map -> <k2, v2> ->
>
> +combine -> <k2, v2> ->
>
> +reduce -> <k3, v3> (output)
>
> +
>
> +== HDFS Architecture
>
> +
>
> +HDFS has a master/slave architecture. An HDFS cluster consists of:
>
> +
>
>
> Better to include the Architecture diagram ( from below link), if it's
> already included, please ignore
>
>
> https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html
> ------------------------------
>
> In content/Hadoop/src/main/asciidoc/index.adoc
> <https://github.com/apache/incubator-training/pull/63#discussion_r363568719>
> :
>
> > +== HDFS Backup node
>
> +
>
> +Backup node is an extension to the Checkpoint node.
>
> +In addition to checkpointing it also receives a stream of edits from the NameNode.
>
> +It maintains its own in-memory copy of the namespace, which is always in sync with the active NameNode namespace state.
>
> +Only one Backup node may be registered with the NameNode at once.
>
> +
>
> +== HDFS Commands
>
> +
>
> +All HDFS commands are invoked by the bin/hdfs script.
>
> +Running the hdfs script without any arguments prints the description for all commands.
>
> +
>
> +Usage: hdfs [SHELL_OPTIONS] COMMAND [GENERIC_OPTIONS] [COMMAND_OPTIONS]
>
> +
>
> +
>
> +== Common HDFS Commands
>
>
> Same we can give for YARN and MR also..?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-training/pull/63?email_source=notifications&email_token=AA5KNGY5L5BOL3FKSVSYFW3Q5BAWVA5CNFSM4JKQDJYKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCQ2LDWQ#pullrequestreview-338997722>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5KNG7SQPGWKIA3GNXAQJDQ5BAWVANCNFSM4JKQDJYA>
> .
>

RyanSkraba commented:
Thanks for the contribution!  This looks encouraging.

greatgautam commented:
Hi,
I am new to the JIRA flow. Since the PR is merged, when can we close the
following issue in JIRA:

https://issues.apache.org/jira/browse/TRAINING-27


thanks,
Gautam

On Fri, May 29, 2020 at 1:20 AM RyanSkraba <notifications@github.com> wrote:

> Thanks for the contribution! This looks encouraging.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-training/pull/63#issuecomment-635841562>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5KNG3COKVJKQILBOJGFFLRT5V6PANCNFSM4JKQDJYA>
> .
>
